datasource.go

```diff
@@ -351,6 +351,32 @@ func (c *Client) DataSourceByUID(uid string) (*DataSource, error) {
 	return result, err
 }
 
+// DataSourceIDByName returns the Grafana data source ID by name.
+func (c *Client) DataSourceIDByName(name string) (int64, error) {
+	path := fmt.Sprintf("/api/datasources/id/%s", name)
+
+	result := struct {
+		ID int64 `json:"id"`
+	}{}
+	err := c.request("GET", path, nil, nil, &result)
+	if err != nil {
+		return result.ID, err
+	}
+
+	return result.ID, err
```

Comment:
Hi! Thanks for your contribution. Here I believe it would be better to return different things if there was an error or not:
```suggestion

	err := c.request("GET", path, nil, nil, &result)
	if err != nil {
		return 0, err
	}

	return result.ID, nil
```